Lib/multiprocessing/pool.py

```diff
@@ -767,12 +772,14 @@ def next(self, timeout=None):
                 item = self._items.popleft()
             except IndexError:
                 if self._index == self._length:
+                    self._pool = None
                     raise StopIteration from None
                 self._cond.wait(timeout)
                 try:
                     item = self._items.popleft()
                 except IndexError:
                     if self._index == self._length:
+                        self._pool = None
```

Comment:
I don't think we can get here without _pool already being removed.

Reply:
This was left from when I was testing against a race. Thanks for the catch!